stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -178,6 +178,8 @@ mul!(C::AbstractMatrix, A::AbstractVecOrMat, B::AbstractVecOrMat) = generic_matm
     rmul!(A, B)
 
 Calculate the matrix-matrix product ``AB``, overwriting `A`, and return the result.
+Here, `B` must have triangular/diagonal structure, which is additionally encoded
```

Comment:
Maybe use a formulation that is more open since e.g. also orthogonal matrices can be applied in-place.

Reply:
Maybe something like "Must have special structure, which is additionally encoded ... and which allows for in-place multiplication."? I realized in https://github.com/JuliaLang/julia/pull/31285#issuecomment-471977636 that we may extend the in-place multiplication to a couple of factorizations involving triangular and orthogonal matrices, basically building on what we have already. Do you think that's of interest to be included in `LinearAlgebra`?